core/src/data/clientGeoJsonSource.cpp

```diff
@@ -117,30 +117,52 @@ void ClientGeoJsonSource::addPoly(const Properties& _tags, const std::vector<Coo
     m_generation++;
 }
 
-std::shared_ptr<TileData> ClientGeoJsonSource::parse(const TileTask& _task,
-                                                     const MapProjection& _projection) const {
-
-    auto data = std::make_shared<TileData>();
+bool ClientGeoJsonSource::process(const TileTask& _task,
+                                  const MapProjection& _projection,
+                                  TileDataSink& _sink) const {
 
     geojsonvt::Tile tile;
     {
         std::lock_guard<std::mutex> lock(m_mutexStore);
-        if (!m_store) { return nullptr; }
+        //if (!m_store) { return nullptr; }
+        if (!m_store) { return false; }
         tile = m_store->getTile(_task.tileId().z, _task.tileId().x, _task.tileId().y);
     }
 
-    Layer layer(""); // empty name will skip filtering by 'collection'
+    // empty name will skip filtering by 'collection'
+    // TODO why not use DataSource name as layer name?
```

Comment:
The scene file already filters data by source name, so it would be redundant to perform filtering against the same name for the layer.